lib/src/main/kotlin/org/dotenv/vault/DotenvVault.kt

```diff
@@ -9,11 +9,18 @@ import org.dotenv.vault.DotenvVault.Companion.loadVault
 class DotenvVault(val dotenv: DotenvVaultAdapter) : Dotenv {
     companion object {
         fun loadVault(key: String?, config: Configuration): Dotenv {
+            // fix: this code should be more intelligent. it should attempt to load using_vault only if it makes sense to:
+            // 1. .env.vault is present
+            // OR
+            // 2. DOTENV_KEY is present
+            // see logic inside of using_vault? method in ruby: https://github.com/dotenv-org/dotenv-vault-ruby/blob/master/lib/dotenv-vault.rb#L27
```

Comment:
So this is actually what it does despite not having an explicit if here

If no vault file is found the dotenv-kotlin library call there will throw DotenvException("file not found")
If no environment key is found the other line will throw DotenvException("cannot find key entry ...")

But I agree that decryptVault method name is misleading, as the actual decryption doesn't happen until both of those things have been initialized correctly. 
I will add and explicit check for key sooner.

The other check as whether the file exists or not, I didn't want to introduce any filesystem api code, since the base kotlin lib, actually the java lib, is already doing that, and it has code for both android and java, which this library would need to replicate, because it can't be extended from the original project.